admin_pages/general_settings/OrganizationSettings.php

```diff
@@ -467,10 +489,90 @@ public function process($form_data = array())
         $this->registry->CFG->currency = new EE_Currency_Config(
             $this->organization_config->CNT_ISO
         );
+        /** @var EE_Country $country */
+        $country = EEM_Country::instance()->get_one_by_ID($this->organization_config->CNT_ISO);
+        if ($country instanceof EE_Country) {
+            $country->set('CNT_active', 1);
+            $country->save();
+            $this->getCountrySubRegions($country);
+        }
         return true;
     }
 
 
+    /**
+     * @param EE_Country $country
+     * @return string
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     */
+    private function getCountrySubRegions(EE_Country $country)
+    {
+        $CNT_ISO = $country->ID();
+        $has_sub_regions = EEM_State::instance()->count(array(array('Country.CNT_ISO' => $CNT_ISO)));
+        if ($has_sub_regions) {
+            return;
+        }
+        $json = $this->getCountryJson(
+            'https://raw.githubusercontent.com/astockwell/countries-and-provinces-states-regions/master/countries.json'
+        );
+        if (is_array($json)) {
+            foreach ($json as $country) {
+                if ($country->code === $CNT_ISO) {
+                    if($country->filename !== null) {
+                        $sub_regions = $this->getCountryJson(
+                            'https://raw.githubusercontent.com/astockwell/countries-and-provinces-states-regions/master/countries/'
```

Comment:
Are we sure this is a good resource to use?  It doesn't appear [to be maintained](https://github.com/astockwell/countries-and-provinces-states-regions/issues/15) and the last time it was updated was 4 years ago...  Other than that, nifty.

Reply:
An alternative, would be for us to fork this and maintain it ourselves.  With it being a critical source for data in EE it might be useful for us to do that?

Reply:
ya I thought about that too... what would be cool about that is that we could also add to the data as well, and include other things like cities, etc

Reply:
forked to: https://github.com/eventespresso/countries-and-provinces-states-regions